### PR TITLE
Introduce workflows for automated deployment of components

### PR DESCRIPTION
**Pull request by denbite**
* implement `deploy-to-testnet` workflow that deploys components to testnet contract
* implement `deploy-to-mainnet` workflow that deploys components to mainnet contract
* provide step-by-step guide for using both workflows in Readme

### CODE REVIEW

**Comment by denbite** [comment]
this perfectly solves https://github.com/frol/bos-component-ts-starter/issues/2

File: .github/workflows/deploy-to-mainnet.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Install bos CLI
+        run: |
+          curl --proto '=https' --tlsv1.2 -LsSf https://github.com/FroVolod/bos-cli-rs/releases/download/v0.3.2/bos-cli-installer.sh | sh
+
+      - name: Deploy widgets
+        run: |
+          which bos
+          echo $PATH
+          cd .bos/transpiled && bos components deploy "$ACCOUNT_ID" sign-as "$ACCOUNT_ID" network-config "$NETWORK" sign-with-plaintext-private-key --signer-public-key "$ACCOUNT_PUBLIC_KEY" --signer-private-key "$ACCOUNT_PRIVATE_KEY" send
```
Comment (frol) [inline .github/workflows/deploy-to-mainnet.yml]:
```suggestion
      - name: Checkout repository
        uses: actions/checkout@v4

      - name: Install dependencies
        run: npm ci

      - name: Build
        run: npm run build

      - name: Deploy widgets
        run: |
          npm run deploy -- "$ACCOUNT_ID" sign-as "$ACCOUNT_ID" network-config "$NETWORK" sign-with-plaintext-private-key --signer-public-key "$ACCOUNT_PUBLIC_KEY" --signer-private-key "$ACCOUNT_PRIVATE_KEY" send
```

File: .github/workflows/deploy-to-mainnet.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+env:
+  ACCOUNT_ID: ${{ vars.MAINNET_ACCOUNT_ID }}
+  ACCOUNT_PUBLIC_KEY: ${{ vars.MAINNET_ACCOUNT_PUBLIC_KEY }}
+  ACCOUNT_PRIVATE_KEY: ${{ secrets.MAINNET_ACCOUNT_PRIVATE_KEY }}
+  NETWORK: mainnet
+
+jobs:
+  deploy-widgets:
+    runs-on: ubuntu-latest
+    name: Deploy widgets to $ACCOUNT_ID
```
Comment (frol) [inline .github/workflows/deploy-to-mainnet.yml]: I feel that $ACCOUNT_ID won't be picked up here. Please, check it on your fork and attach a link with a successful deployment 🙏

**Review by frol — CHANGES_REQUESTED** [review]

File: .github/workflows/deploy-to-mainnet.yml (hunk 2 above)
Comment (denbite) [reply .github/workflows/deploy-to-mainnet.yml]: you're right, it doesn't, I've just removed it
https://github.com/denbite/bos-component-ts-starter/actions/runs/9259579272/job/25471752485

**Review by frol — APPROVED** [review]

**Pull request merged**